src/rust/ensogl/lib/text/src/component/area.rs

```diff
@@ -903,6 +916,11 @@ impl AreaModel {
         clipboard::write_text(encoded);
     }
 
+    fn cut(&self, selections:&[String]) {
+        self.copy(selections);
+        self.buffer.frp.delete_left();
```

Comment:
No function in model should emit frp events if it does not have to. Move this line inside of the FRP logic you have created. Something like:
```
eval  sels_cut ((s) m.cut(s));
eval_ sels_cut (m. buffer.delete_left());
```